### DISCUSSION TITLE
Copied Layout from docs which Using View Transitions API to change appearance transformation didn't work on mine

### DISCUSSION THREAD

**Discussion by fr1g** [vuejs/vitepress#4012] [Q&A]
As my title, I just copied the example code from https://vitepress.dev/guide/extending-default-theme#on-appearance-toggle and this is not working on my project.
some hints:
- I have custom CSS, and a css-transition registered on '*' but I think this is not the root of issue
- inside my copied Layout.vue, says 'No property on "document" was named "startViewTransition"'
- I just updated to vitepress to 1.2.3, before that I was in 1.0.0-43 or what I forgot
- This, perhaps written right, but not working on my browser(126.0.2592.81 MS-Edge Chromium, Windows x64)

If there's anything else required to seek and solve the problem, please let me know🙏

**Comment by brc-dd** [comment]
Can you provide a minimal, reproducible example using https://vitepress.new.

That example works fine - https://stackblitz.com/edit/vite-kftjgn?file=docs%2F.vitepress%2Ftheme%2Findex.ts,docs%2F.vitepress%2Ftheme%2FLayout.vue

> inside my copied Layout.vue, says 'No property on "document" was named "startViewTransition"'

this might be because you haven't installed the types - https://www.npmjs.com/package/@types/dom-view-transitions -- install that and reference it from global.d.ts/env.d.ts - you can just `// @ts-ignore` that too.

> This, perhaps written right, but not working on my browser(126.0.2592.81 MS-Edge Chromium, Windows x64)

Most likely your code is wrong. That browser supports view transition api.

**Comment by shellRaining** [comment]
@fr1g You can refer to my code, which might help you~

https://github.com/shellRaining/blog/blob/b751cdfac3d7b3ad205e0aac226c67d4f436628a/theme/Home/AppearanceSwitcher.vue

**Comment by fr1g** [comment]
thanks, I‘ll check it up later!Отправлено с iPhone16 июля 2024 г., в 23:50, shellRaining ***@***.***> написал(а):﻿
@fr1g You can refer to my code, which might help you~
https://github.com/shellRaining/blog/blob/b751cdfac3d7b3ad205e0aac226c67d4f436628a/theme/Home/AppearanceSwitcher.vue
